packages/analytics-js-plugins/src/xhrQueue/index.ts

```diff
@@ -1,14 +1,156 @@
 /* eslint-disable no-param-reassign */
-import { ExtensionPlugin, PluginName, ApplicationState } from '../types/common';
+import { QUEUE_NAME, REQUEST_TIMEOUT_MS } from './constants';
+import { XHRQueueItem } from './types';
+import {
+  getNormalizedQueueOptions,
+  getDeliveryPayload,
+  validatePayloadSize,
+  getDeliveryUrl,
+} from './utilities';
+import {
+  ExtensionPlugin,
+  PluginName,
+  ApplicationState,
+  DoneCallback,
+  HttpClient,
+  IErrorHandler,
+  ILogger,
+  QueueOpts,
+  RudderEvent,
+  IHttpClient,
+} from '../types/common';
+import { Queue, getCurrentTimeFormatted, toBase64 } from '../utilities/common';
 
 const pluginName = PluginName.XhrQueue;
 
+let eventsQueue: Queue;
+let writeKey: string;
+let dataplaneUrl: string;
+let httpClient: IHttpClient;
+let logger: ILogger | undefined;
+let errorHandler: IErrorHandler | undefined;
+let initialized = false;
+
 const XhrQueue = (): ExtensionPlugin => ({
   name: pluginName,
   deps: [],
   initialize: (state: ApplicationState) => {
     state.plugins.loadedPlugins.value = [...state.plugins.loadedPlugins.value, pluginName];
   },
+  xhrDeliveryQueue: {
+    init(state: ApplicationState, inErrorHandler?: IErrorHandler, inLogger?: ILogger): void {
+      if (initialized) {
+        return;
+      }
+
+      errorHandler = inErrorHandler;
+      logger = inLogger;
+      dataplaneUrl = state.lifecycle.activeDataplaneUrl.value as string;
+      writeKey = state.lifecycle.writeKey.value as string;
+
+      httpClient = new HttpClient(errorHandler, logger);
+      httpClient.setAuthHeader(writeKey);
+
+      const finalQOpts = getNormalizedQueueOptions(
+        state.loadOptions.value.queueOptions as QueueOpts,
+      );
+
+      eventsQueue = new Queue(
```

Comment:
you can create the queue through the storage manager. there is an extension point there for that I believe

Reply:
I couldn't find any way to create the queue from the store/storage manager. Could you please clarify?

Reply:
I was reffering to this.pluginManager?.invokeMultiple('queuestore.create', this.setStore); of store manager but this is to create new Store not queues. 

Maybe I need to think if some refactoring is needed in localstorage-retry package to use this when creating new stores.

Unrelated to you PR, please ignore